### PR TITLE
Xrefs and wording updates for modules manual

### PR DESCRIPTION
**Pull request by kshyatt**
Not sure if this is the best way to phrase the role of `stdlib` but I think it's more accurate than what we had. Lots of links added!

### CODE REVIEW

**Review by vchuravy — APPROVED** [review]

File: doc/src/manual/modules.md [hunk 1]
```diff
@@ -116,24 +116,32 @@ end
...
+`Main` is the top-level module, and Julia starts with `Main` set as the current module.  Variables
+defined at the prompt go in `Main`, and [`varinfo()`](@ref) lists variables in `Main`.
 
-Core contains all identifiers considered "built in" to the language, i.e. part of the core language
+`Core` contains all identifiers considered "built in" to the language, i.e. part of the core language
 and not libraries. Every module implicitly specifies `using Core`, since you can't do anything
 without those definitions.
 
-Base is the standard library (the contents of base/). All modules implicitly contain `using Base`,
+`Base` is part of the standard library (the contents of `base/`). All modules implicitly contain `using Base`,
```
Comment (fredrikekre) [inline doc/src/manual/modules.md]: I think we should differentiate between `Base` and stdlib.

Comment (kshyatt) [reply doc/src/manual/modules.md]: Agreed. Any suggestions?

Comment (kshyatt) [reply doc/src/manual/modules.md]: ... Bueller?

File: doc/src/manual/modules.md [hunk 2]
```diff
@@ -117,25 +117,33 @@ end
...
-Base is a module that contains basic functionality (the contents of base/). All modules implicitly contain `using Base`,
+`Base` is a module that contains basic functionality (the contents of `base/`). All modules implicitly contain `using Base`,
 since this is needed in the vast majority of cases.
 
+Other modules live in `stdlib/`, and may be loaded as needed by users as they would load
+their own custom modules or packages. For example, if you wanted to perform some eigenvalue
+calculations (the tooling for which is part of `stdlib`), you could use:
+
+```
+using IterativeEigenSolvers
```
Comment (fredrikekre) [inline doc/src/manual/modules.md]: This is not a thing anymore. Maybe `Test`?

File: doc/src/manual/modules.md [hunk 3]
```diff
@@ -117,25 +117,33 @@ end
...
 
-Base is a module that contains basic functionality (the contents of base/). All modules implicitly contain `using Base`,
+`Base` is a module that contains basic functionality (the contents of `base/`). All modules implicitly contain `using Base`,
 since this is needed in the vast majority of cases.
 
+Other modules live in `stdlib/`, and may be loaded as needed by users as they would load
+their own custom modules or packages. For example, if you wanted to perform some eigenvalue
+calculations (the tooling for which is part of `stdlib`), you could use:
+
+```
```
Comment (fredrikekre) [inline doc/src/manual/modules.md]:
````
```julia
````

File: doc/src/manual/modules.md [hunk 4]
```diff
@@ -117,25 +117,33 @@ end
...
-Core contains all identifiers considered "built in" to the language, i.e. part of the core language
+`Core` contains all identifiers considered "built in" to the language, i.e. part of the core language
 and not libraries. Every module implicitly specifies `using Core`, since you can't do anything
 without those definitions.
 
-Base is a module that contains basic functionality (the contents of base/). All modules implicitly contain `using Base`,
+`Base` is a module that contains basic functionality (the contents of `base/`). All modules implicitly contain `using Base`,
 since this is needed in the vast majority of cases.
 
+Other modules live in `stdlib/`, and may be loaded as needed by users as they would load
```
Comment (fredrikekre) [inline doc/src/manual/modules.md]: IMO where they live is irrelevant. Maybe something like
```
Julia also comes with some standard library modules ...
```

**Comment by ViralBShah** [comment]
@kshyatt Hope you don't mind me pinging on these old PRs, but this looks pretty good and the conflicts look easy to resolve. Would it be possible to update this PR?

**Comment by kshyatt** [comment]
@ViralBShah working on it now

**Comment by kshyatt** [comment]
OK, there were no conflicts, just needed a rebase. Other than that, is there anything we need to fix?

**Comment by ViralBShah** [comment]
Looks great. Can we rebase to master? Please merge it when it passes tests as well!

**Comment by ViralBShah** [comment]
I guess this needs updating:
```
��� Warning: no doc found for reference '[`cfunction`](@ref)' in src/manual/modules.md.
��� @ Documenter.CrossReferences /buildworker/worker/doctest_linux64/build/doc/deps/packages/Documenter/PLD7m/src/CrossReferences.jl:160
[ Info: CheckDocument: running document checks.
```

**Comment by ViralBShah** [comment]
Updated `cfunction`->`@cfunction`

File: doc/src/manual/modules.md [hunk 5]
```diff
@@ -122,6 +122,16 @@ There are three important standard modules:
 * [`Base`](@ref) contains basic functionality that is useful in almost all cases.
 * [`Main`](@ref) is the top-level module and the current module, when Julia is started.
 
+
+!!! note "Standard library modules"
+    By default Julia ships with some standard library modules. These behave like regular
+    Julia packages except that you don't need to install them explicitly. For example,
+    if you wanted to perform some unit testing, you could load the `Test` standard library
+    as follows:
```
Comment (KristofferC) [inline doc/src/manual/modules.md:130]: This is slightly misleading since in package you do need to "install" them by adding them with the package manager.

**Pull request merged**